[URIUtils][CURL] Refactor URLParsing tests as parametrized tests

CrystalP opened the pull request:
## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here. -->

Rework the new url parsing unit tests as parametrized tests to make the results easier to understand

There is no unit test 0.json, so that number is skipped in order to align the iteration number and the file number in the gtest output, and avoid a potentially confusing offset of 1.

## Motivation and context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

Inability to investigate the failures.

## How has this been tested?
<!--- Please describe in detail how you tested your change. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

Ran the tests :-)
They're now reported individually.

## What is the effect on users?
<!--- Summarize the effect of this change on Kodi end-users. -->
<!--- If the PR does not have a noticeable impact (e.g., if it only changes documentation), -->
<!--- just leave it empty. Put in more detail the bigger the impact is. -->
<!--- This section may be used for automatic creation of release notes. -->

none

## Screenshots (if appropriate):

### Before:

E:\Devpt\kodi-2023\xbmc\utils\test\TestUrlParsing.cpp(446): error: Expected equality of these values:
  param.expectedGetShareName
    Which is: ""
  curl.GetShareName()
    Which is: "/foo/bar.apk/bob.avi"
E:\Devpt\kodi-2023\xbmc\utils\test\TestUrlParsing.cpp(446): error: Expected equality of these values:
  param.expectedGetShareName
    Which is: ""
  curl.GetShareName()
    Which is: "/foo/bar.apk/secondary/apkfile.apk/bob.avi"
[  FAILED  ] TestURLParseDetails.ParseAllURLResults (2908 ms)
[----------] 1 test from TestURLParseDetails (2909 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (3367 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] TestURLParseDetails.ParseAllURLResults

 1 FAILED TEST

### After:

[ RUN      ] TestURLParse/TestURLParseDetails.RunTest/748_json
[       OK ] TestURLParse/TestURLParseDetails.RunTest/748_json (3 ms)
[----------] 749 tests from TestURLParse/TestURLParseDetails (3399 ms total)

[----------] Global test environment tear-down
[==========] 749 tests from 1 test suite ran. (3752 ms total)
[  PASSED  ] 646 tests.
[  FAILED  ] 103 tests, listed below:
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/3_json, where GetParam() = 3
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/4_json, where GetParam() = 4
...
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/718_json, where GetParam() = 718
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/719_json, where GetParam() = 719

103 FAILED TESTS

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [X] I have added tests to cover my change
- [ ] All new and existing tests passed

CrystalP committed:
[URIUtils][CURL] Refactor parsing tests as parametrized tests

thexai commented:
In case someone is not aware, failure is no only in WIN-64, others also failing but continues with tests failed:

e.g. WIN-32
```
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/155, where GetParam() = 155
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/156, where GetParam() = 156
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/157, where GetParam() = 157
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/158, where GetParam() = 158
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/159, where GetParam() = 159
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/160, where GetParam() = 160
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/161, where GetParam() = 161
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/654, where GetParam() = 654
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/655, where GetParam() = 655
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/656, where GetParam() = 656
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/657, where GetParam() = 657
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/662, where GetParam() = 662
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/663, where GetParam() = 663
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/664, where GetParam() = 664
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/665, where GetParam() = 665
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/666, where GetParam() = 666
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/667, where GetParam() = 667
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/668, where GetParam() = 668
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/669, where GetParam() = 669
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/670, where GetParam() = 670
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/671, where GetParam() = 671
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/672, where GetParam() = 672
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/673, where GetParam() = 673
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/674, where GetParam() = 674
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/684, where GetParam() = 684
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/685, where GetParam() = 685
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/700, where GetParam() = 700
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/701, where GetParam() = 701
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/702, where GetParam() = 702
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/703, where GetParam() = 703
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/708, where GetParam() = 708
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/712, where GetParam() = 712
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/713, where GetParam() = 713
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/714, where GetParam() = 714
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/715, where GetParam() = 715
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/716, where GetParam() = 716
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/717, where GetParam() = 717
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/718, where GetParam() = 718
[  FAILED  ] TestURLParse/TestURLParseDetails.RunTest/719, where GetParam() = 719

105 FAILED TESTS
  YOU HAVE 5 DISABLED TESTS

Clean shutdown of TestGlobalPattern1
        1 file(s) moved.
Done running testsuite
------------------------------------------------------------
New run name is 'xbmc-PR27461-4189bd5-#39842'
[File exists] check if file exists [gtestresults.xml]
Run condition [File exists] preventing perform for step [[Archive the artifacts, Publish xUnit test result report]]
SSH: Skipping [Mirrors] - Label [true] does not match expression [false]
[MSBuild] Parsing console log (workspace: 'c:\jenkws\workspace\WIN-32')
[MSBuild] -> found 57 issues (skipped 33 duplicates)
[MSBuild] Parsing console log (workspace: 'c:\jenkws\workspace\WIN-32')
The recommended git tool is: NONE
using credential github-app-xbmc
 > git rev-parse "4189bd5263d710fcbc476d9daec13277e0dc9df0^{commit}" # timeout=10
The recommended git tool is: NONE
using credential github-app-xbmc
 > git rev-parse "4189bd5263d710fcbc476d9daec13277e0dc9df0^{commit}" # timeout=10
[MSBuild] Post processing issues on 'Win-Gotham' with source code encoding 'UTF-8'
[MSBuild] Creating SCM blamer to obtain author and commit information for affected files
[MSBuild] -> Git blamer successfully created in working tree 'C:/jenkws/workspace/WIN-32'
[MSBuild] Resolving file names for all issues in workspace 'c:\jenkws\workspace\WIN-32'
[MSBuild] -> resolved paths in source directory (8 found, 10 not found)
[MSBuild] Resolving module names from module definitions (build.xml, pom.xml, or Manifest.mf files)
[MSBuild] -> resolved module names for 57 issues
[MSBuild] Resolving package names (or namespaces) by parsing the affected files
[MSBuild] -> resolved package names of 18 affected files
[MSBuild] Applying 1 filters on the set of 57 issues (38 issues have been removed, 19 issues will be published)
[MSBuild] Creating fingerprints for all affected code blocks to track issues over different builds
[MSBuild] -> created fingerprints for 19 issues (skipped 0 issues)
[MSBuild] Invoking Git blamer to create author and commit information for 3 affected files
[MSBuild] -> GIT_COMMIT env = '4189bd5263d710fcbc476d9daec13277e0dc9df0'
[MSBuild] -> Git commit ID = '4189bd5263d710fcbc476d9daec13277e0dc9df0'
[MSBuild] -> Git working tree = 'C:/jenkws/workspace/WIN-32'
[MSBuild] -> blamed authors of issues in 3 files
[MSBuild] Blaming of authors took 3 seconds
[MSBuild] Copying affected files to Jenkins' build folder '/var/jenkins_home/jobs/WIN-32/builds/39842/files-with-issues'
[MSBuild] -> 3 copied, 0 not in workspace, 0 not-found, 0 with I/O error
[MSBuild] Skipping cleaning of source code files in old builds
[MSBuild] Repository miner is not configured, skipping repository mining
[MSBuild] Successfully parsed console log
[MSBuild] -> found 57 issues (skipped 33 duplicates)
[MSBuild] Parsing console log (workspace: 'c:\jenkws\workspace\WIN-32')
[MSBuild] Post processing issues on 'Win-Gotham' with source code encoding 'UTF-8'
[MSBuild] Creating SCM blamer to obtain author and commit information for affected files
[MSBuild] -> Git blamer successfully created in working tree 'C:/jenkws/workspace/WIN-32'
[MSBuild] Resolving file names for all issues in workspace 'c:\jenkws\workspace\WIN-32'
[MSBuild] -> resolved paths in source directory (8 found, 10 not found)
[MSBuild] Resolving module names from module definitions (build.xml, pom.xml, or Manifest.mf files)
[MSBuild] -> resolved module names for 57 issues
[MSBuild] Resolving package names (or namespaces) by parsing the affected files
[MSBuild] -> resolved package names of 18 affected files
[MSBuild] Applying 1 filters on the set of 57 issues (38 issues have been removed, 19 issues will be published)
[MSBuild] Creating fingerprints for all affected code blocks to track issues over different builds
[MSBuild] -> created fingerprints for 19 issues (skipped 0 issues)
[MSBuild] Invoking Git blamer to create author and commit information for 3 affected files
[MSBuild] -> GIT_COMMIT env = '4189bd5263d710fcbc476d9daec13277e0dc9df0'
[MSBuild] -> Git commit ID = '4189bd5263d710fcbc476d9daec13277e0dc9df0'
[MSBuild] -> Git working tree = 'C:/jenkws/workspace/WIN-32'
[MSBuild] -> blamed authors of issues in 3 files
[MSBuild] Blaming of authors took 3 seconds
[MSBuild] Copying affected files to Jenkins' build folder '/var/jenkins_home/jobs/WIN-32/builds/39842/files-with-issues'
[MSBuild] -> 3 copied, 0 not in workspace, 0 not-found, 0 with I/O error
[MSBuild] Skipping cleaning of source code files in old builds
[MSBuild] Repository miner is not configured, skipping repository mining
The recommended git tool is: NONE
using credential github-app-xbmc
 > git rev-parse "4189bd5263d710fcbc476d9daec13277e0dc9df0^{commit}" # timeout=10
The recommended git tool is: NONE
using credential github-app-xbmc
[MSBuild] Reference build recorder is not configured
[MSBuild] No valid reference build found
[MSBuild] All reported issues will be considered outstanding
[MSBuild] No quality gates have been set - skipping
[MSBuild] Health report is disabled - skipping
[MSBuild] Created analysis result for 19 issues (found 0 new issues, fixed 0 issues)
[MSBuild] Attaching ResultAction with ID 'msbuild' to build 'WIN-32 #39842'.
[Checks API] No suitable checks publisher found.
Started calculate disk usage of build
Finished Calculation of disk usage of build in 0 seconds
Started calculate disk usage of workspace
Finished Calculation of disk usage of workspace in  21 second
Finished: SUCCESS
```

This is a different issue....   "Jenkins continues with FAILED tests (and not disabled)"

fuzzard commented:
Arm can't be run on x64 host, uwp has never run tests.
X86 I forgot about, and honestly don't really care to go fix atm

78andyp reviewed (approved):
One small comment - otherwise looks good to me.
Thanks.

CrystalP commented:
Yes could have used a const but the intermediate variable is not really necessary so I removed it.
Fixed a comment typo.
Renamed run_parse_test to RunParseTest to follow guidelines.
There is more renaming to do but the functions are not involved in the scope of the PR so I left them alone.
